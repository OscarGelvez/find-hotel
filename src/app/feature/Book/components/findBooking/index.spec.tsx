import '@testing-library/jest-dom';

import { cleanup, render } from '@testing-library/react';
import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { FindBookingForm } from './index';

afterEach(cleanup);

it('rederizar el componente FindBookingForm sin errores', () => {
  const div = document.createElement('div');
  ReactDOM.render(<FindBookingForm onFindBooking={() => null} />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renderiza el componente FindBookingForm y verifica existencia de componentes hijos', () => {
  const { getByTestId } = render(
    <FindBookingForm onFindBooking={() => null} />
  );

  expect(getByTestId('find-bookings')).toContainElement(
    getByTestId('form-find-bookings')
  );

  expect(getByTestId('form-find-bookings')).toHaveFormValues({
    email: '',
    identification_type: '',
    identification: '',
  });

  expect(getByTestId('form-find-bookings')).toContainElement(
    getByTestId('form-bookings-email')
  );
  expect(getByTestId('form-find-bookings')).toContainElement(
    getByTestId('form-bookings-identification_type')
  );
  expect(getByTestId('form-find-bookings')).toContainElement(
    getByTestId('form-bookings-identification')
  );

  expect(getByTestId('form-find-bookings')).toContainElement(
    getByTestId('form-bookings-btn-submit')
  );
  expect(getByTestId('form-find-bookings')).toContainElement(
    getByTestId('form-bookings-btn-reset')
  );

  expect(getByTestId('form-bookings-btn-submit')).toHaveAttribute(
    'type',
    'submit'
  );

  expect(getByTestId('form-bookings-btn-reset')).toHaveAttribute(
    'type',
    'reset'
  );
});

it('Compara snapshots del componente FindBookingForm', () => {
  const element = renderer
    .create(<FindBookingForm onFindBooking={() => null} />)
    .toJSON();
  expect(element).toMatchSnapshot();
});
