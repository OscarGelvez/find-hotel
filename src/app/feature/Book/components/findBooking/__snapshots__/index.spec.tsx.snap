// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compara snapshots del componente FindBookingForm 1`] = `
<div
  className="card-rooms"
  data-testid="find-bookings"
>
  <div
    className="row"
  >
    <div
      className="card filter-area "
    >
      <h4
        className="mt-3"
      >
        Encuentra tu reserva
      </h4>
      <div
        className="col"
      >
        <form
          action="#"
          data-testid="form-find-bookings"
          onReset={[Function]}
          onSubmit={[Function]}
        >
          <div
            className="form-group row mb-3 mt-3"
          >
            <div
              className=" col-12 col-md-4"
            >
              <label
                className="col-12 col-form-label"
                htmlFor="email"
              >
                Correo electrónico*
              </label>
              <input
                className="form-control col-12"
                data-testid="form-bookings-email"
                name="email"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="eje: joe@email.com"
                type="email"
                value=""
              />
            </div>
            <div
              className="col-12 col-md-4"
            >
              <label
                className="col-12 col-form-label"
                htmlFor="identification_type"
              >
                Tipo de identificación*
              </label>
              <select
                aria-label=".form-select"
                className=" d-block w-100"
                data-testid="form-bookings-identification_type"
                id="identification_type"
                name="identification_type"
                onBlur={[Function]}
                onChange={[Function]}
                value=""
              >
                <option
                  value=""
                >
                  Seleccione
                </option>
                <option
                  value="1"
                >
                  Cédula de ciudadanía
                </option>
                <option
                  value="2"
                >
                  Cédula de extranjeria
                </option>
                <option
                  value="3"
                >
                  Nit
                </option>
                <option
                  value="4"
                >
                  Pasaporte
                </option>
              </select>
            </div>
            <div
              className=" col-12 col-md-4"
            >
              <label
                className="col-12 col-form-label"
                htmlFor="identification"
              >
                Identificación*
              </label>
              <input
                className="form-control"
                data-testid="form-bookings-identification"
                id="identification"
                name="identification"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="eje: 60830147"
                value=""
              />
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="mt-4 text-center"
            >
              <button
                className="sc-AxjAm cYJbNq btn w-75"
                data-testid="form-bookings-btn-submit"
                type="submit"
              >
                Buscar
              </button>
            </div>
            <div
              className="text-center mb-2 mt-1"
            >
              <button
                className="sc-AxjAm cYJbNq button-link"
                data-testid="form-bookings-btn-reset"
                type="reset"
              >
                limpiar
              </button>
            </div>
          </div>
        </form>
      </div>
    </div>
  </div>
</div>
`;
